The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7197619269/job/19605170511) [(plain)](https://github.com/rust-lang/rust/commit/8465e405a3c3c5a36c08e8a6806f91f90574c511/checks/19605170511/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=ouz-a
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_9c512d28-5eaa-450c-8ae8-54c9e261747d
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=add_emit_stable_mir_tests
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_9c512d28-5eaa-450c-8ae8-54c9e261747d
GITHUB_REF=refs/pull/118375/merge
GITHUB_REF_NAME=118375/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=58735e080e46edc502f48b9ab2b723d48c9b06ad
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_9c512d28-5eaa-450c-8ae8-54c9e261747d
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_9c512d28-5eaa-450c-8ae8-54c9e261747d
GITHUB_TRIGGERING_ACTOR=ouz-a
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/118375/merge
GITHUB_WORKFLOW_SHA=58735e080e46edc502f48b9ab2b723d48c9b06ad
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
Built container sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Looks like docker image is the same as before, not uploading
https://ci-caches.rust-lang.org/docker/7ebc15c01a233894034d277c8cce4e949f4e7791f66b4727c8fb6e058a0b8171d6152e1441d677cef0653843ceeee469c097b8699b2bb74249e674f6aa1a8813
sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Wed Dec 13 15:45:56 UTC 2023
  network time: Wed, 13 Dec 2023 15:45:56 GMT
  network time: Wed, 13 Dec 2023 15:45:56 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-missing-tools', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: rust.codegen-units-std := 1
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
62  | |         StatementKind::FakeRead(_, _) => format!("StatementKind::FakeRead:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:62:42
   --> compiler/stable_mir/src/mir/pretty.rs:62:42
    |
62  |         StatementKind::FakeRead(_, _) => return format!("StatementKind::FakeRead:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
64  | |             format!("StatementKind::SetDiscriminant:Unimplemented")
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:64:13
   --> compiler/stable_mir/src/mir/pretty.rs:64:13
    |
64  |             return format!("StatementKind::SetDiscriminant:Unimplemented");

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
66  | |         StatementKind::Deinit(_) => format!("StatementKind::Deinit:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:66:37
   --> compiler/stable_mir/src/mir/pretty.rs:66:37
    |
66  |         StatementKind::Deinit(_) => return format!("StatementKind::Deinit:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
67  | |         StatementKind::StorageLive(_) => format!("StatementKind::StorageLive:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:67:42
   --> compiler/stable_mir/src/mir/pretty.rs:67:42
    |
67  |         StatementKind::StorageLive(_) => return format!("StatementKind::StorageLive:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
68  | |         StatementKind::StorageDead(_) => format!("StatementKind::StorageDead:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:68:42
   --> compiler/stable_mir/src/mir/pretty.rs:68:42
    |
68  |         StatementKind::StorageDead(_) => return format!("StatementKind::StorageDead:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
69  | |         StatementKind::Retag(_, _) => format!("StatementKind::Retag:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:69:39
   --> compiler/stable_mir/src/mir/pretty.rs:69:39
    |
69  |         StatementKind::Retag(_, _) => return format!("StatementKind::Retag:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
70  | |         StatementKind::PlaceMention(_) => format!("StatementKind::PlaceMention:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:70:43
   --> compiler/stable_mir/src/mir/pretty.rs:70:43
    |
70  |         StatementKind::PlaceMention(_) => return format!("StatementKind::PlaceMention:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
72  | |             format!("StatementKind::AscribeUserType:Unimplemented")
    | |             ------------------------------------------------------- in this macro invocation
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:72:13
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
74  | |         StatementKind::Coverage(_) => format!("StatementKind::Coverage:Unimplemented"),
...   |
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:74:39
   --> compiler/stable_mir/src/mir/pretty.rs:74:39
    |
74  |         StatementKind::Coverage(_) => return format!("StatementKind::Coverage:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
75  | |         StatementKind::Intrinsic(_) => format!("StatementKind::Intrinsic:Unimplemented"),
    | |                                        ------------------------------------------------- in this macro invocation
76  | |         StatementKind::ConstEvalCounter => format!("StatementKind::ConstEvalCounter:Unimplemented"),
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:75:40
   --> compiler/stable_mir/src/mir/pretty.rs:75:40
    |
75  |         StatementKind::Intrinsic(_) => return format!("StatementKind::Intrinsic:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
76  | |         StatementKind::ConstEvalCounter => format!("StatementKind::ConstEvalCounter:Unimplemented"),
    | |                                            -------------------------------------------------------- in this macro invocation
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:76:44
   --> compiler/stable_mir/src/mir/pretty.rs:76:44
    |
76  |         StatementKind::ConstEvalCounter => return format!("StatementKind::ConstEvalCounter:Unimplemented");,

error[E0308]: mismatched types
   --> /checkout/library/alloc/src/macros.rs:126:9
    |
---
    |
   ::: compiler/stable_mir/src/mir/pretty.rs:56:5
    |
56  | /     match statement {
57  | |         StatementKind::Assign(place, rval) => {
58  | |             pretty.push_str(format!("        _{} = ", place.local).as_str());
59  | |             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
...   |
77  | |         StatementKind::Nop => format!("StatementKind::Nop:Unimplemented"),
78  | |     }
    | |_____- expected this to be `()`
    |
help: you might have meant to return this value
help: you might have meant to return this value
   --> compiler/stable_mir/src/mir/pretty.rs:77:31
    |
77  |         StatementKind::Nop => return format!("StatementKind::Nop:Unimplemented");,

For more information about this error, try `rustc --explain E0308`.
error: could not compile `stable_mir` (lib) due to 12 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

--- a/compiler/stable_mir/src/mir/pretty.rs
+++ b/compiler/stable_mir/src/mir/pretty.rs
@@ -58,18 +58,35 @@ pub fn pretty_statement(statement: &StatementKind) -> String {
             pretty.push_str(format!("        _{} = ", place.local).as_str());
             pretty.push_str(format!("{}", &pretty_rvalue(rval)).as_str());
         }
-        StatementKind::FakeRead(_, _) => todo!(),
-        StatementKind::SetDiscriminant { .. } => todo!(),
-        StatementKind::Deinit(_) => todo!(),
-        StatementKind::StorageLive(_) => todo!(),
-        StatementKind::StorageDead(_) => todo!(),
-        StatementKind::Retag(_, _) => todo!(),
-        StatementKind::PlaceMention(_) => todo!(),
-        StatementKind::AscribeUserType { .. } => todo!(),
-        StatementKind::Coverage(_) => todo!(),
-        StatementKind::Intrinsic(_) => todo!(),
-        StatementKind::ConstEvalCounter => (),
-        StatementKind::Nop => (),
+        // FIXME: Add rest of the statements
+        StatementKind::FakeRead(_, _) => {
+            return String::from("StatementKind::FakeRead:Unimplemented");
+        }
+        StatementKind::SetDiscriminant { .. } => {
+            return String::from("StatementKind::SetDiscriminant:Unimplemented");
+        }
+        StatementKind::Deinit(_) => return String::from("StatementKind::Deinit:Unimplemented"),
+        StatementKind::StorageLive(_) => {
+            return String::from("StatementKind::StorageLive:Unimplemented");
+        }
+        StatementKind::StorageDead(_) => {
+            return String::from("StatementKind::StorageDead:Unimplemented");
+        }
+        StatementKind::Retag(_, _) => return String::from("StatementKind::Retag:Unimplemented"),
+        StatementKind::PlaceMention(_) => {
+            return String::from("StatementKind::PlaceMention:Unimplemented");
+        }
+        StatementKind::AscribeUserType { .. } => {
+            return String::from("StatementKind::AscribeUserType:Unimplemented");
+        }
+        StatementKind::Coverage(_) => return String::from("StatementKind::Coverage:Unimplemented"),
+        StatementKind::Intrinsic(_) => {
+            return String::from("StatementKind::Intrinsic:Unimplemented");
+        }
+        StatementKind::ConstEvalCounter => {
+            return String::from("StatementKind::ConstEvalCounter:Unimplemented");
+        }
+        StatementKind::Nop => return String::from("StatementKind::Nop:Unimplemented"),
     }
     pretty
 }
@@ -355,7 +372,7 @@ pub fn pretty_rvalue(rval: &Rvalue) -> String {
             pretty.push_str(" ");
             pretty.push_str(&pretty_ty(cnst.ty().kind()));
         }
-        Rvalue::ShallowInitBox(_, _) => todo!(),
+        Rvalue::ShallowInitBox(_, _) => (),
         Rvalue::ThreadLocalRef(item) => {
             pretty.push_str("thread_local_ref");
             pretty.push_str(format!("{:#?}", item).as_str());

--- a/src/tools/tidy/src/ui_tests.rs
+++ b/src/tools/tidy/src/ui_tests.rs
@@ -11,7 +11,7 @@ use std::path::{Path, PathBuf};
 const ENTRY_LIMIT: usize = 900;
 // FIXME: The following limits should be reduced eventually.
 const ISSUES_ENTRY_LIMIT: usize = 1852;
-const ROOT_ENTRY_LIMIT: usize = 866;
+const ROOT_ENTRY_LIMIT: usize = 867;
 
 const EXPECTED_TEST_FILE_EXTENSIONS: &[&str] = &[
     "rs",     // test source files

--- a/tests/ui/stable-mir-print/basic_function.rs
+++ b/tests/ui/stable-mir-print/basic_function.rs
@@ -1,0 +1,15 @@
+// compile-flags: -Z unpretty=stable-mir -Z mir-opt-level=3
+// check-pass
+// only-x86_64
+
+fn foo(i:i32) -> i32 {
+    i + 1
+}
+
+fn bar(vec: &mut Vec<i32>) -> Vec<i32> {
+    let mut new_vec = vec.clone();
+    new_vec.push(1);
+    new_vec
+}
+
+fn main(){}

--- a/tests/ui/stable-mir-print/basic_function.stdout
+++ b/tests/ui/stable-mir-print/basic_function.stdout
@@ -1,0 +1,234 @@
+// WARNING: This is highly experimental output it's intended for stable-mir developers only.
+// If you find a bug or want to improve the output open a issue at https://github.com/rust-lang/project-stable-mir.
+fn foo(_0: i32) -> i32 {
+    let mut _0: (i32, bool);
+}
+    bb0: {
+        _2 = 1 Add const 1_i32
+        assert(!move _2 bool),"attempt to compute `{} + {}`, which would overflow", 1, const 1_i32) -> [success: bb1, unwind continue]
+    }
+    bb1: {
+        _0 = move _2
+        return
+    }
+fn bar(_0: &mut Ty {
+    id: 10,
+    kind: RigidTy(
+        Adt(
+            AdtDef(
+                DefId {
+                    id: 3,
+                    name: "std::vec::Vec",
+                },
+            ),
+            GenericArgs(
+                [
+                    Type(
+                        Ty {
+                            id: 11,
+                            kind: Param(
+                                ParamTy {
+                                    index: 0,
+                                    name: "T",
+                                },
+                            ),
+                        },
+                    ),
+                    Type(
+                        Ty {
+                            id: 12,
+                            kind: Param(
+                                ParamTy {
+                                    index: 1,
+                                    name: "A",
+                                },
+                            ),
+                        },
+                    ),
+                ],
+            ),
+        ),
+    ),
+}) -> Ty {
+    id: 10,
+    kind: RigidTy(
+        Adt(
+            AdtDef(
+                DefId {
+                    id: 3,
+                    name: "std::vec::Vec",
+                },
+            ),
+            GenericArgs(
+                [
+                    Type(
+                        Ty {
+                            id: 11,
+                            kind: Param(
+                                ParamTy {
+                                    index: 0,
+                                    name: "T",
+                                },
+                            ),
+                        },
+                    ),
+                    Type(
+                        Ty {
+                            id: 12,
+                            kind: Param(
+                                ParamTy {
+                                    index: 1,
+                                    name: "A",
+                                },
+                            ),
+                        },
+                    ),
+                ],
+            ),
+        ),
+    ),
+} {
+    let mut _0: Ty {
+    id: 10,
+    kind: RigidTy(
+        Adt(
+            AdtDef(
+                DefId {
+                    id: 3,
+                    name: "std::vec::Vec",
+                },
+            ),
+            GenericArgs(
+                [
+                    Type(
+                        Ty {
+                            id: 11,
+                            kind: Param(
+                                ParamTy {
+                                    index: 0,
+                                    name: "T",
+                                },
+                            ),
+                        },
+                    ),
+                    Type(
+                        Ty {
+                            id: 12,
+                            kind: Param(
+                                ParamTy {
+                                    index: 1,
+                                    name: "A",
+                                },
+                            ),
+                        },
+                    ),
+                ],
+            ),
+        ),
+    ),
+};
+    let mut _1: &Ty {
+    id: 10,
+    kind: RigidTy(
+        Adt(
+            AdtDef(
+                DefId {
+                    id: 3,
+                    name: "std::vec::Vec",
+                },
+            ),
+            GenericArgs(
+                [
+                    Type(
+                        Ty {
+                            id: 11,
+                            kind: Param(
+                                ParamTy {
+                                    index: 0,
+                                    name: "T",
+                                },
+                            ),
+                        },
+                    ),
+                    Type(
+                        Ty {
+                            id: 12,
+                            kind: Param(
+                                ParamTy {
+                                    index: 1,
+                                    name: "A",
+                                },
+                            ),
+                        },
+                    ),
+                ],
+            ),
+        ),
+    ),
+};
+    let _2: ();
+    let mut _3: &mut Ty {
+    id: 10,
+    kind: RigidTy(
+        Adt(
+            AdtDef(
+                DefId {
+                    id: 3,
+                    name: "std::vec::Vec",
+                },
+            ),
+            GenericArgs(
+                [
+                    Type(
+                        Ty {
+                            id: 11,
+                            kind: Param(
+                                ParamTy {
+                                    index: 0,
+                                    name: "T",
+                                },
+                            ),
+                        },
+                    ),
+                    Type(
+                        Ty {
+                            id: 12,
+                            kind: Param(
+                                ParamTy {
+                                    index: 1,
+                                    name: "A",
+                                },
+                            ),
+                        },
+                    ),
+                ],
+            ),
+        ),
+    ),
+};
+}
+    bb0: {
+        _3 = refShared1
+        _2 = const <Vec<i32> as Clone>::clone(move _3) -> [return: bb1, unwind continue]
+    }
+    bb1: {
+        _5 = refMut {
+    kind: TwoPhaseBorrow,
+}2
+        _4 = const Vec::<i32>::push(move _5, const 1_i32) -> [return: bb2, unwind: bb3]
+    }
+    bb2: {
+        _0 = move _2
+        return
+    }
+    bb3: {
+        drop(_2) -> [return: bb4, unwind terminate]
+    }
+    bb4: {
+        resume
+    }
+fn main() -> () {
+}
+    bb0: {
+        return
+    }

